### ISSUE TITLE
[BUG] avoid-collection-methods-with-unrelated-types does not support null keys

### ISSUE THREAD

**Issue by maxlapides** [dart-code-checker/dart-code-metrics#784]
* **Dart code metrics version: 4.13.0**
* **Dart sdk version: 2.16.2**

The new `avoid-collection-methods-with-unrelated-types` seems to be mishandling this edge case:

```dart
final myMap = <int?, String>{null: 'hello'};
final myStr = myMap[null];
```

This should be allowed because `null` is of type `int?`

**Comment by incendial** [comment]
Available in 4.14.0 🚀
